platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,493 @@
+# TSS - Roster Proposal
+
+---
+
+## Summary
+
+This proposal outlines the design specification of rosters to support Threshold Signature Scheme (TSS) block signing.
+It details the roster's lifecycle, data storage, API, DevOps workflow, and changes necessary changes to associated
+components within the platform.
+
+|      Metadata      |                   Entities                    |
+|--------------------|-----------------------------------------------|
+| Designers          | Kore, Anthony, Ed, Austin                     |
+| Functional Impacts | TSS, Address Book, Reconnect,                 |
+| Related Proposals  | TSS-Library, TSS-Ledger-Id, TSS-Block-Signing |
+| HIPS               | N/A                                           |
+
+## Purpose and Context
+
+The introduction of the Threshold Signature Scheme (TSS) requires a new mechanism for managing node participation in
+consensus and block signing.
+The Roster, a subset of the address book, will provide the data structure and API for this mechanism.
+This proposal provides a specification for the behavior of rosters starting from their creation from the Address Book to
+their terminal state.
+A roster reaches a terminal state when it is either adopted by the platform or replaced by a new roster.
+The candidate roster will always get adopted on the next software upgrade.
+
+The Hedera App (henceforth reffered to as 'App') will maintain a version of the Address Book as a dynamic list that
+reflects the desired future state of
+the
+network's nodes. This dynamic list of Addresses will be continuously updated by HAPI transactions, such as those that
+create, update, or delete nodes.
+At some point when the App decides it's time to begin work on adopting a new address book (scope beyond this proposal),
+it will create a `Roster` object with information from the Address Book and pass it to the platform.
```

Comment:
Style note: Can we get newlines between paragraphs? Or if these are part of one paragraph, remove the newlines at the end of each sentence?

Reply:
I'll make the paragraph changes. For removing newlines, our code style has a max line length I tried to adapt to. Perhaps we can make an exception for design docs?

Reply:
Max line-length is good. But clearly there is something else going on, otherwise you wouldn't have lines like 29, 30, and 31.